account_bank_receipt/models/account_bank_receipt.py

```diff
@@ -0,0 +1,378 @@
+# Copyright 2019 Ecosoft Co., Ltd (http://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import _, api, fields, models
+import odoo.addons.decimal_precision as dp
+from odoo.exceptions import ValidationError, UserError
+
+
+class AccountBankReceipt(models.Model):
+    _name = "account.bank.receipt"
+    _description = "Account Bank Receipt"
+    _order = 'receipt_date desc'
+
+    @api.depends('company_id', 'currency_id', 'receipt_line')
+    def _compute_bank_receipt(self):
+        for deposit in self:
+            total = 0.0
+            count = 0
+            reconcile = False
+            currency_none_same_company_id = False
+            if deposit.company_id.currency_id != deposit.currency_id:
+                currency_none_same_company_id = deposit.currency_id.id
+            for line in deposit.receipt_line:
+                count += 1
+                total += line.allocation
+            if deposit.move_id:
+                for line in deposit.move_id.line_ids:
+                    if line.debit > 0 and line.reconciled:
+                        reconcile = True
+            deposit.total_amount = total
```

Comment:
Please adjust all vairables, we should not have word "deposit" or "check" anymore?